src/test/java/io/supertokens/test/WebserverTest.java

```diff
@@ -791,30 +792,42 @@ public void invalidBasePathTest() throws InterruptedException, IOException {
             Utils.setValueInConfig("base_path", "somepath/");
             String[] args = { "../" };
             TestingProcess process = TestingProcessManager.start(args);
-            EventAndException e = process.checkOrWaitForEvent(PROCESS_STATE.INIT_FAILURE);
-            assertTrue(e != null && e.exception instanceof QuitProgramException
-                    && e.exception.getMessage().equals("base_path must start with a '/'"));
-            Utils.reset();
+            EventAndException e = process.checkOrWaitForEvent(PROCESS_STATE.STARTED);
+            assertEquals(Config.getConfig(process.main).getBasePath(), "/somepath");
+            process.kill();
         }
-
         {
             Utils.setValueInConfig("base_path", "/somepath/");
             String[] args = { "../" };
             TestingProcess process = TestingProcessManager.start(args);
-            EventAndException e = process.checkOrWaitForEvent(PROCESS_STATE.INIT_FAILURE);
-            assertTrue(e != null && e.exception instanceof QuitProgramException
-                    && e.exception.getMessage().equals("base_path cannot end with '/'"));
-            Utils.reset();
+            EventAndException e = process.checkOrWaitForEvent(PROCESS_STATE.STARTED);
+            assertEquals(Config.getConfig(process.main).getBasePath(), "/somepath");
+            process.kill();
+        }
+        {
+            Utils.setValueInConfig("base_path", "somepath");
+            String[] args = { "../" };
+            TestingProcess process = TestingProcessManager.start(args);
+            EventAndException e = process.checkOrWaitForEvent(PROCESS_STATE.STARTED);
+            assertEquals(Config.getConfig(process.main).getBasePath(), "/somepath");
+            process.kill();
+        }
+        {
+            Utils.setValueInConfig("base_path", "/somepath");
+            String[] args = { "../" };
+            TestingProcess process = TestingProcessManager.start(args);
+            EventAndException e = process.checkOrWaitForEvent(PROCESS_STATE.STARTED);
+            assertEquals(Config.getConfig(process.main).getBasePath(), "/somepath");
+            process.kill();
         }
-
         {
             Utils.setValueInConfig("base_path", "/some path");
             String[] args = { "../" };
             TestingProcess process = TestingProcessManager.start(args);
             EventAndException e = process.checkOrWaitForEvent(PROCESS_STATE.INIT_FAILURE);
             assertTrue(e != null && e.exception instanceof QuitProgramException
                     && e.exception.getMessage().equals("Invalid characters in base_path config"));
-            Utils.reset();
+            process.kill();
```

Comment:
- In this case, we expect the core to not start, so don't need to call kill.
- It's a good idea to call ` Utils.reset();` in each of the cases above as that resets the config file to its original form.

Reply:
Since `Utils.reset();` is included in the `beforeEach` method, shouldn't we just delete this line so it's not called twice?

Reply:
Yea. We could. But calling it twice is not an issue. I was there originally only so that it matched the "format" of the other test cases in this one test case. But we can remove it.. yea